tests/jsmarionette/client/marionette-client/lib/marionette/client.js

```diff
@@ -681,22 +622,19 @@
     waitForSync: function(test, callback, interval, timeout) {
       var err, result;
 
-      function done(_err, _result) {
-        err = _err;
-        result = _result;
-      }
-
-      function sleep(waitMillis) {
-        setTimeout(marionetteScriptFinished, waitMillis);
-      }
-
       while (Date.now() < timeout) {
         if (err || result) {
           return callback(err);
         }
 
-        test(done);
-        this.executeAsyncScript(sleep, [interval]);
+        test(function(_err, _result) {
```

Comment:
This should also fail jshint because you shouldn't define functions within loops. Do you need to modify `waitForSync`?